src/MatrixFields/field_name_dict.jl

```diff
@@ -181,11 +181,51 @@ function get_internal_entry(
         entry
     elseif name_pair[2] == @name() && broadcasted_has_field(T, name_pair[1])
         # multiplication case 2 or 4, second argument
-        Base.broadcasted(entry) do matrix_row
-            map(matrix_row) do matrix_row_entry
-                broadcasted_get_field(matrix_row_entry, name_pair[1])
-            end
-        end # Note: This assumes that the entry is in a FieldMatrixBroadcasted.
+        target_field_eltype = broadcasted_get_field_type(T, name_pair[1])
+        if target_field_eltype <: Number
+            T_band = eltype(entry)
+            singleton_datalayout =
+                DataLayouts.singleton(Fields.field_values(entry))
+            # BandMatrixRow with same lowest diagonal and bandwidth as `entry`, with a scalar eltype
+            scalar_band_type = BandMatrixRow{
+                T_band.parameters[1],
+                T_band.parameters[2],
+                eltype(parent(entry)),
+            }
+            field_dim_size = DataLayouts.ncomponents(Fields.field_values(entry))
+            scalar_field_offset = get_field_first_index_offset(
+                name_pair[1],
+                target_field_eltype,
+                T,
+            )
+            band_element_size = Int(div(sizeof(T), sizeof(target_field_eltype)))
```

Comment:
```suggestion
            band_element_size = div(sizeof(T), sizeof(target_field_eltype))
```
`div` already returns an `Int`, so you shouldn't need to convert it